router/forwarder.go

```diff
@@ -351,16 +292,7 @@ func (fwd *Forwarder) flush() {
 	err := fwd.udpSender.Send(fwd.enc.Bytes())
 	if err != nil {
 		if mtbe, ok := err.(MsgTooBigError); ok {
```

Comment:
It's a shame to have this in the `Forwarder` code given that we can only get `MsgTooBigError` in `ForwarderDF`.

How about we increase the scope of of the custom handler slightly...

````
if err := processSendError(fwd.udpSender.Send(fwd.enc.Bytes())); err != nil && PosixError(err) != syscall.ENOBUFS {
    fwd.conn.Shutdown(err)
}
````
`Forwarder` would supply an identify function for `processSendError`, whereas `ForwarderDF` would supply a function that handles `MsgTooBigError`.